src/audio/dsm/sof_dsm.c

```diff
@@ -0,0 +1,231 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
```

Comment:
is this a copy-paste or did you really intend to use GPL2.0+ for firmware? I am not sure this is compatible with the directions to use a permissive license in all other public SOF parts?

Reply:
Actually I copied license part from the reference code, smart_amp_test.c file [https://github.com/thesofproject/sof/blob/master/src/audio/smart_amp_test.c](url)
Since I created smart_amp_maxim.c based on above file. I kept same license.
All the information in this commit is open source based and the firmware is not included.

Reply:
@ryans-lee https://github.com/thesofproject/sof/blob/af136e0325c7cd45c2be456928a7bd6e9d8cda76/src/audio/smart_amp_test.c#L1 shows BSD-3-clause in the SPDX line.

Reply:
Thanks for pointing this out. Shall change GPL2.0+ to BSD-3-Clause.

> @ryans-lee
> 
> https://github.com/thesofproject/sof/blob/af136e0325c7cd45c2be456928a7bd6e9d8cda76/src/audio/smart_amp_test.c#L1
> 
> shows BSD-3-clause in the SPDX line.

Thanks for pointing this out. Shall change GPL2.0+ to BSD-3-Clause.